### PR TITLE
Update PR template

### PR DESCRIPTION
**Pull request by victorlin**
## Description of proposed changes

A few small changes to improve the PR authoring and reviewing experience.

## Related issue(s)

N/A

## Checklist

- [x] Checks pass

### CODE REVIEW

**Review by tsibley — COMMENTED** [review]
I think we need to consider the question, "Who is the PR template for?"

If it's primarily for us, e.g. because we open the most PRs ourselves, then +1 for this change or similar changes as we should be streamlining the process of PR description usefulness for ourselves.

If it's primarily for external contributors, as I think it started out as during the pandemic for [ncov](https://github.com/nextstrain/ncov), then I'm not sure we want to ditch the hints and extra guidance.

**Comment by victorlin** [comment]
@tsibley That's a good point. Since we welcome external contributions, I think the PR template should be for external contributors. It's easy for us to delete/ignore the hints that we don't need.

I've repurposed this PR to make a few smaller changes to the template.

File: .github/pull_request_template.md
Comment (victorlin) [inline .github/pull_request_template.md:1]: As an aside, I looked into whether we could use multiple PR templates. I was thinking there might be some sort of "template picker" such as the one for issues. While there is [support for multiple PR templates](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository) (using the `.github/PULL_REQUEST_TEMPLATE` folder),  there's no "template picker" for PRs as of now. The only way to use templates in that folder is through [URL query parameters](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/using-query-parameters-to-create-a-pull-request) (e.g. `https://github.com/octo-org/octo-repo/compare/main...my-branch?quick_pull=1&template=issue_template.md`).

Comment (tsibley) [reply .github/pull_request_template.md:1]: Ah, yeah, I should have mentioned I also looked into that before commenting. Came to the same conclusion. Drats.

**Review by tsibley — APPROVED** [review]

**Pull request merged**